src/specialized.jl

```diff
@@ -120,19 +120,32 @@ function parse_string(ps::MemoryParserState, b::IOBuffer)
     b
 end
 
-function parse_number(pc::ParserContext, ps::MemoryParserState)
+function parse_number(pc::ParserContext{<:Any,<:Any,AllowNanInf}, ps::MemoryParserState) where AllowNanInf
     s = p = ps.s
     e = length(ps)
     isint = true
+    negative = false
+
+    @inbounds c = ps[p]
+
+    # Parse and keep track of initial minus sign (for parsing -Infinity)
+    if AllowNanInf && c == MINUS_SIGN
+        negative = true
+        p += 1
```

Comment:
I can't test this right now but I suspect that this breaks parsing regular negative floats in AllowNanInf mode. I might be wrong but the code seems to read like it skips the minus sign here, even if not followed by I. Could you add a test for that?

Reply:
This one was actually ok (`s` above marks the start of the number).

Reply:
(Not broken, but tests added.)